orttraining/orttraining/core/graph/graph_augmenter.cc

```diff
@@ -17,8 +17,14 @@ Status GraphAugmenter::AugmentGraph(Graph& graph, const GraphDefs& graph_element
     }
   }
 
-  // Add new nodes to the graph.
-  for (const auto& node_def : graph_element_defs.NodeDefs()) {
+  // Add new nodes to the graph, sorted by name to get a consistent ordering.
```

Comment:
In which case is the ordering important? Is it related to seeding or this is an opportunistic fix?

Reply:
More of an opportunistic fix, to ensure all distributed processes execute same sequence.  I actually don't think it's causing issues here, since from any given tensor's perspective it's computed with the same order of operations.  I think I'm going to split this change out, and measure its effects separately.